lib/fizzy/parser_expr.cpp

```diff
@@ -546,7 +546,10 @@ parser_result<Code> parse_expr(
         case Instr::i64_store32:
         {
             // alignment
-            std::tie(std::ignore, pos) = leb128u_decode<uint32_t>(pos, end);
+            uint32_t align;
+            std::tie(align, pos) = leb128u_decode<uint32_t>(pos, end);
+            if ((1 << static_cast<uint64_t>(align)) > metrics.memory_width)
```

Comment:
why cast is needed?

Reply:
My compiler was complaining that shifting `1 << uint32_t` can't be represented in `uint32_t`, for some reason it picked up the type from `align`. Setting the type on `1` would be nicer I guess.

However, the compiler is not complaining anymore so can remove.

Reply:
Nicer way is doing it as `uint64_t{1} << align` (or `uint32_t{1} << align`, depending on needs).